persistence/modeshape-persistence-file/src/main/java/org/modeshape/persistence/file/FileDb.java

```diff
@@ -102,13 +102,13 @@ public Document get( String key ) {
 
     @Override
     public List<SchematicEntry> load( Collection<String> keys ) {
+        final TransactionStore.TransactionMap<String, Document> txContent = transactionalContent(false);
+        final TransactionStore.TransactionMap<String, Document> actualContent = txContent != null ? txContent : persistedContent; 
         return keys.stream()
-                   .map(persistedContent::get)
+                   .map(actualContent::get)
                    .filter(Objects::nonNull)
-                   .map(document -> {
-                       SchematicEntry schematicEntry = () -> document;
-                       return schematicEntry;
-                   }).collect(Collectors.toList()); 
+                   .map(document -> (SchematicEntry) () -> document)
```

Comment:
Ok, scratch that. Took me a minute that you're not casting but instead are creating a `SchematicEntry` using this technique. Since that operation is done in several places, it's probably useful to add a static method to `SchematicEntry` that is something like:

```
public static SchematicEntry fromDocument(Document source) {
    return ()->source;
}
```

Then the above line could be written:

```
.map(SchematicEntry::fromDocument)
```

which I think is more readable.